File: ekart-backend/Order-Service/src/main/java/com/ekart/order/service/PromoCodesService.java
package com.ekart.order.service;

import java.util.List;

import org.springframework.stereotype.Service;
import org.springframework.web.bind.annotation.PathVariable;

import com.ekart.order.entity.PromoCodes;

@Service
public interface PromoCodesService {
	
	public double getDiscountPrice(@PathVariable String promoCode);
	
	public PromoCodes savePromocode(PromoCodes promoCode);
	
	public List<PromoCodes> getAllPromoCodes();

	public List<PromoCodes> getAllPromoCodesByStatus(String status);

}

File: ui/ui-frontend/src/Services/PromoCodesService.js
import { authenticateApi } from "./axiosConfig";

export const PromoCodesService = {
    async getAllPromoCodes() {
        const res = await authenticateApi.request({
            url: `/api/promocodes/getAllPromoCodes`,
            method: "GET",
        });
        return res;
    },

    async getDiscountPrice(promoCode) {
        const res = await authenticateApi.request({
            url: `/api/promocodes/getDiscountPrice/${promoCode}`,
            method: "GET",
        });
        return res;
    },

    async getAllPromoCodesByStatus(status) {
        const res = await authenticateApi.request({
            url: `/api/promocodes/getAllPromoCodes/${status}`,
            method: "GET",
        });
        return res;
    },
};
